### ISSUE TITLE
Loss becomes NaN during training on Sleep-EDF 2018 with default SleePyCo code

### ISSUE THREAD

**Issue by maniupiaha** [gist-ailab/SleePyCo#20]
Hi,
Thank you for sharing your great work! I’m trying to reproduce the results of SleePyCo using the Sleep-EDF 2018 dataset. I used the original code without any modifications, but during training, I encountered an issue where the loss becomes NaN after a few epochs, which causes the model performance to drop significantly.

I followed the preprocessing and training instructions as described in the repository.
Do you have any suggestions on what might be causing this, or how to resolve the issue?

Thanks in advance!

**Comment by SeongjuLee** [comment]
Thank you for your interest in our work!
Did you encounter an issue where the loss becomes NaN during the CRL phase or MTCL phase?
I recommend modifying the code from...
```
73            self.optimizer.zero_grad()
74            loss.backward()
75            self.optimizer.step()
```
to
```
73            self.optimizer.zero_grad()
74            loss.backward()
++            torch.nn.utils.clip_grad_norm_(self.model.parameters(), 5.0)
75            self.optimizer.step()
```
This is the example of modification in `train_crl.py`

**Comment by SeongjuLee** [comment]
Feel free to leave an issue if you have any further questions. Thanks!
